### DISCUSSION TITLE
Duplicate/cloned columns when generating table with mongodb query

### DISCUSSION THREAD

**Discussion by GabrielSalernoILARI** [getredash/redash#6961] [Q&A] [answered]
Whenever I try to query the mongodb database that i connected to redash, using a query like this:

{
    "collection": "DATABASE_NORMALIZZATO",
    "aggregate": [
        {
            "$project": {
                "_id": "$_id",
                "Tipologia prodotto": "$DESCRIZIONE AREA",
                "Nome prodotto": "$DESCRIZIONE BENE DA OSPEDALE",
                "Importo": "$IMPORTO (NO IVA)"
            }
        }
    ]
}

I get a giant table that, instead of having only 3 columns, has 3 columns FOR EACH ROW. e.g. the table has 4 results (4 rows, 4 documents in the db), the table header will look like this:

_id
Tipologia prodotto
Nome prodotto
Importo
_id
Tipologia prodotto
Nome prodotto
Importo
_id
Tipologia prodotto
Nome prodotto
Importo
_id
Tipologia prodotto
Nome prodotto
Importo

So i have clones of the columns which repeat the same data in each row. e.g. the first row has its id shown multiple times (as many as there are rows)

**Comment by GabrielSalernoILARI** [comment, marked as answer]
I am a complete idiot.
You simply have to add the "fields".
this is how the query looks now:

```
{
    "collection": "DATABASE_NORMALIZZATO",
    "fields":{
        "_id": 1,
        "TotalProducts": 1
    },
    "aggregate": [
        {
            "$group": {
                "_id": "$COSTRUTTORE | TITOLARE",
                "TotalProducts": {
                    "$sum": 1
                }
            }
        },
        {
            "$sort": [{
			"name": "TotalProducts",
			"direction": -1
		    }]
        }
    ]
}
```

and of course it works. I completely forgot you have to do that, lol. Reminder for anyone who might come across this: "fields" is like projection in mongodb.

**Reply by justinclift** [reply]
That's good info for anyone else that has the same problem.  I've just tweaked the formatting of that to be a bit more readable too, in case it helps. :smile:
